controllers/dataprotection/utils.go

```diff
@@ -149,58 +149,103 @@ func GetTargetPods(reqCtx intctrlutil.RequestCtx,
 	if !existPodSelector(selector) {
 		return nil, nil
 	}
-	labelSelector, err := metav1.LabelSelectorAsSelector(selector.LabelSelector)
-	if err != nil {
-		return nil, err
-	}
-	pods := &corev1.PodList{}
-	if err = cli.List(reqCtx.Ctx, pods,
-		client.InNamespace(reqCtx.Req.Namespace),
-		client.MatchingLabelsSelector{Selector: labelSelector}); err != nil {
-		return nil, err
+
+	listPodsByLabel := func(labelSelector *metav1.LabelSelector) (*corev1.PodList, error) {
+		if labelSelector == nil {
+			return nil, intctrlutil.NewNotFound("not found labelSelector")
+		}
+		selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+		if err != nil {
+			return nil, err
+		}
+		pods := &corev1.PodList{}
+		err = cli.List(reqCtx.Ctx, pods,
+			client.InNamespace(reqCtx.Req.Namespace),
+			client.MatchingLabelsSelector{Selector: selector})
+		return pods, err
 	}
 
-	if len(pods.Items) == 0 {
-		return nil, fmt.Errorf("failed to find target pods by backup policy %s/%s",
-			backupPolicy.Namespace, backupPolicy.Name)
+	pods, err := listPodsByLabel(selector.LabelSelector)
+	if err != nil {
+		return nil, err
 	}
-	sort.Sort(intctrlutil.ByPodName(pods.Items))
+	findPodErr := fmt.Errorf("failed to find target pods by backup policy %s/%s",
+		backupPolicy.Namespace, backupPolicy.Name)
 	var targetPods []*corev1.Pod
 	if len(selectedPodNames) == 0 || backupType == dpv1alpha1.BackupTypeContinuous {
 		switch selector.Strategy {
 		case dpv1alpha1.PodSelectionStrategyAny:
 			// always selecting the first pod
 			pod := dputils.GetFirstIndexRunningPod(pods)
+			if pod == nil {
+				pods, err := listPodsByLabel(selector.AlternateLabelSelector)
+				if err != nil && !intctrlutil.IsNotFound(err) {
+					return nil, err
+				}
+				if err == nil {
+					pod = dputils.GetFirstIndexRunningPod(pods)
+				}
+			}
 			if pod != nil {
 				targetPods = append(targetPods, pod)
 			}
 		case dpv1alpha1.PodSelectionStrategyAll:
+			allAvailable := true
+			for i := range pods.Items {
+				if !intctrlutil.IsAvailable(&pods.Items[i], 0) {
```

Comment:
Why would the availability of pods trigger a switch to using the alternativeLabelSelector?

Reply:
This behavior looks unreasonable

Reply:
Other places don't have similar logic that uses the IsAvailable condition to filter pods.